version: '2.12.2'
services:
    postgres:
        restart: always
        ports:
        - "6032:5432"
        environment:
            - POSTGRES_USER=toto
            - POSTGRES_PASSWORD=toto
            - POSTGRES_DB=postgres
        networks:
            - odoo
        volumes:
            - 'pgdata:/var/lib/postgresql/data'
        container_name: db_odoo
        image: 'postgres:13'
    odoo:
        depends_on:
            - postgres
        ports:
            - '8000:8000'
        container_name: d_odoo
        networks:
            - odoo
        volumes:
            - '$PWD/config:/etc/odoo'
            - '$PWD/addons:/mnt/extra-addons'
            - 'odoo-web-data:/var/lib/odoo'
        environment:
            - USER=toto
            - PASSWORD=toto
            - HOST=db_odoo
        image: odoo:13.0

    ic-webapp:
      image : ic-webapp:1.0
      ports: 
       - 8080:8080
      volumes:
       - type: bind
         source: ./app.py
         target: /data/app.py
      depends_on:
       - pgadmin
       - odoo

    pgadmin:
      image: dpage/pgadmin4
      container_name: pgadmin4_container
      restart: always
      ports:
       - "6081:80"
      networks:
       - odoo
      environment:
       - PGADMIN_DEFAULT_EMAIL=toto@domain.com
       - PGADMIN_DEFAULT_PASSWORD=toto
      volumes:
       - pgadmin-data:/var/lib/pgadmin
      depends_on:
       - postgres
volumes:
    odoo-web-data:
        external: FALSE
    pgdata:
        external: FALSE
    pgadmin-data:
        external: FALSE
networks:
    odoo:
        driver: bridge